catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -753,13 +838,75 @@ public void init() {
           res.body(util.getJson(ImmutableMap.of("message", "Invalid values for numbers")));
         });
 
+    exception(
+        SourceUnavailableException.class,
+        (ex, req, res) -> {
+          res.status(503);
+          res.header(CONTENT_TYPE, APPLICATION_JSON);
+          LOGGER.debug("A source was unavailable when trying to fetch", ex);
+          res.body(
+              util.getJson(
+                  ImmutableMap.of(
+                      "message",
+                      "Could not connect to the sources at this time. Please try again later.")));
+        });
+
     exception(EntityTooLargeException.class, util::handleEntityTooLargeException);
 
     exception(IOException.class, util::handleIOException);
 
     exception(RuntimeException.class, util::handleRuntimeException);
   }
 
+  private SourceInfoResponse getSources() throws SourceUnavailableException {
+    SourceInfoRequestEnterprise sourceInfoRequestEnterprise =
+        new SourceInfoRequestEnterprise(false);
+
+    return catalogFramework.getSourceInfo(sourceInfoRequestEnterprise);
+  }
+
+  private Stream<String> extractPotentialBrokenAssociations(
+      Set<String> idsToCheck, Metacard metacard) {
+    if (!idsToCheck.contains(metacard.getId())) {
+      // Any metacard not on the deleted list got pulled in because its pointing
+      // to something being deleted
+      return Stream.of(metacard.getId());
+    } else {
+      // Metacards on the delete list have their related and derived pulled
+      return Stream.concat(
+          getAttributes(metacard, Metacard.RELATED), getAttributes(metacard, Metacard.DERIVED));
+    }
+  }
+
+  private Filter getParentAndChildrenFilter(String id) {
+    Filter root = util.getFilterBuilder().attribute(Metacard.ID).is().equalTo().text(id);
+    Filter derived = util.getFilterBuilder().attribute(Metacard.DERIVED).is().like().text(id);
+    Filter related = util.getFilterBuilder().attribute(Metacard.RELATED).is().like().text(id);
+
+    Filter parents = util.getFilterBuilder().anyOf(related, derived);
+
+    return util.getFilterBuilder().anyOf(root, parents);
```

Comment:
If this is supposed to be `anyOf`, then why not do `anyOf(root, related, derived)`?

Reply:
done